### PR TITLE
Add `prove_prop` command

### PR DESCRIPTION
**Pull request by chameco**

### CODE REVIEW

**Comment by RyanGlScott** [comment]
At a quick glance, I'm not sure what the difference between this and `prove_print` is, given that they have the same type signatures. Perhaps a test case and a description in the SAW manual would help clarify things.

File: src/SAWScript/Builtins.hs [hunk 1]
```diff
@@ -949,6 +949,26 @@ provePrintPrim script t = do
     InvalidProof _stats _cex pst -> failProof pst
     UnfinishedProof pst -> failProof pst
 
+provePropPrim ::
+  ProofScript () ->
+  TypedTerm ->
+  TopLevel Theorem
+provePropPrim script t = do
```
Comment (andreistefanescu) [inline src/SAWScript/Builtins.hs:956]: let's implement a helper that is parametric in a function with type `Term -> IO Prop`, and then let's implement `provePrintPrim`  and `provePropPrim` by using the helper with `predicateToProp sc Universal` and `termToProp sc`

File: src/SAWScript/Interpreter.hs [hunk 2]
```diff
@@ -1313,6 +1313,15 @@ primitives = Map.fromList
     , "if unsuccessful."
     ]
 
+  , prim "prove_prop"         "ProofScript () -> Term -> TopLevel Theorem"
```
Comment (andreistefanescu) [inline src/SAWScript/Interpreter.hs:1316]: maybe `prove_extcore`?

**Review by andreistefanescu — APPROVED** [review]
👍

**Comment by andreistefanescu** [comment]
I searched for tests for `offline_extcore`, and found nothing. At some point, let's add some tests, including one for this command, but, I don't think this change should be delayed.

**Pull request merged**